compiler/luci/pass/src/DynamicBatchToSingleBatchPass.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/DynamicBatchToSingleBatchPass.h"
+
+#include <luci/IR/CircleNode.h>
+#include <loco.h>
+
+namespace luci
+{
+
+bool DynamicBatchToSingleBatchPass::run(loco::Graph *g)
+{
+  assert(g); // FIX CALLER UNLESS
+
+  bool changed = false;
+
+  auto graph_inputs = g->inputs();
+
+  // Assume the first dimension is batch dimension
+  const uint32_t BATCH_DIM = 0;
+
+  for (auto node : loco::input_nodes(g))
+  {
+    auto input_node = loco::must_cast<luci::CircleInput *>(node);
+
+    if (input_node->rank() == 0)
+      continue;
+
+    // Skip if batch dimension is known
+    if (input_node->dim(BATCH_DIM).known())
+      continue;
+
+    // 'set' will make the dimension known
+    input_node->dim(BATCH_DIM).set(1);
```

Comment:
Is this really valid thing to do?

Reply:
Could you explain your question a little bit more ? This will make dynamic batch to single batch, which is friendly to some backends.

Reply:
I have no knowledge about doing this so the question is as wrote. We had no previous communication about this. It would be helpful if you can share any knowledge with doing like this.
If not, it is also OK, and for this, with various test models with unknown dimension would validate and help understanding what is supported and what is not. As I know, current test models don't have unknown dimension cases.

Reply:
For actual numbers, is it OK to do like this with rank 1 and 2?
For I can think is , some rank 2 model would be HW, some would have other meanings. I'm not sure unknown H would be meaningful and it is OK to fix -1 it to 1.

But, as this is compilation option, users can decide it is valid to apply this option or not.
But from previous experience, I cannot say most users do.

Reply:
Let me explain this PR a bit more. This option will simply change the size of the first dimension to 1, if its size is unknown (= dynamic). I decided to use the "first" dimension, because most models are NHWC or NCHW. Of course some models have different layouts, e.g., HW as you mentioned. In that case, users should not use this option. That is why this option is not included in O1.

Reply:
@seanshpark  As discussed offline, I'll throw an exception if input has unknown dimension, but its rank is not four.